src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartReferenceLinesConfiguration.tsx

```diff
@@ -133,53 +230,42 @@ export default function ChartReferenceLinesConfiguration({
-              otherAxisPosition: Yup.number().test({
-                name: 'otherAxisPosition',
-                message:
-                  type === 'minor'
-                    ? 'Enter a percentage between 0 and 100%'
-                    : `Enter a position within the ${
-                        axis === 'x' ? 'Y' : 'X'
-                      } axis min/max range`,
-                test: (value: number) => {
-                  if (typeof value !== 'number') {
-                    return true;
-                  }
-                  if (type === 'minor') {
-                    return value >= 0 && value <= 100;
-                  }
-                  return minorAxisDomain
-                    ? value >= minorAxisDomain?.min &&
-                        value <= minorAxisDomain.max
-                    : true;
-                },
-              }),
-              position: Yup.string()
-                .required('Enter position')
-                .test({
-                  name: 'axisPosition',
-                  message: `Enter a position within the ${
-                    axis === 'x' ? 'X' : 'Y'
-                  } axis min/max range`,
-                  test: (value: number) => {
-                    return type === 'minor' && minorAxisDomain
-                      ? value >= minorAxisDomain?.min &&
-                          value <= minorAxisDomain.max
-                      : true;
-                  },
-                }),
-              style: Yup.string()
-                .required('Enter style')
-                .oneOf<ReferenceLineStyle>(['dashed', 'solid', 'none']),
-            })}
+            validationSchema={validationSchema}
             onSubmit={(values, helpers) => {
-              onAddLine(values);
+              let updatedValues = produce(values, draft => {
+                delete draft.otherAxisPositionType;
+              });
+
+              // On minor axis remove values for different other axis position types.
+              if (type === 'minor') {
+                if (
+                  values.otherAxisPositionType === otherAxisPositionTypes.custom
+                ) {
+                  updatedValues = produce(updatedValues, draft => {
+                    delete draft.otherAxisEnd;
+                    delete draft.otherAxisStart;
+                  });
+                } else if (
+                  values.otherAxisPositionType ===
+                  otherAxisPositionTypes.betweenDataPoints
+                ) {
+                  updatedValues = produce(updatedValues, draft => {
+                    delete draft.otherAxisPosition;
+                  });
+                } else {
+                  updatedValues = produce(updatedValues, draft => {
+                    delete draft.otherAxisEnd;
+                    delete draft.otherAxisPosition;
+                    delete draft.otherAxisStart;
+                  });
+                }
+              }
```

Comment:
We don't need to run `produce` twice, and it'd also be cleaner to do this as a switch:

```ts
const updatedValues = produce(values, draft => {
  switch (draft.otherAxisPositionType) {
    case otherAxisPositionTypes.custom:
      delete draft.otherAxisStart;
      delete draft.otherAxisEnd;
      break;

    case otherAxisPositionTypes.betweenDataPoints:
      delete draft.otherAxisPosition;
      break;

    case otherAxisPositionTypes.default:
      delete draft.otherAxisEnd;
      delete draft.otherAxisPosition;
      delete draft.otherAxisStart;
      break;
  }

  delete draft.otherAxisPositionType;
});

```

It doesn't seem to me like we need the wrapping condition on the 'minor' type either. If the `otherAxisPosition` type field isn't present, then the above switch statement will just ignore it anyway.